engine/src/main/java/org/hibernate/validator/internal/util/ExecutableParameterNameProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.util;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import javax.validation.ParameterNameProvider;
+
+/**
+ * Allows to obtain parameter names from {@link Executable}s in a uniform fashion. Delegates to the configured
+ * {@link ParameterNameProvider}.
+ *
+ * @author Gunnar Morling
+ */
+public class ExecutableParameterNameProvider {
+
+	private final ParameterNameProvider delegate;
+
+	public ExecutableParameterNameProvider(ParameterNameProvider delegate) {
+		this.delegate = delegate;
+	}
+
+	public List<String> getParameterNames(Executable executable) {
+		if ( executable instanceof Method ) {
```

Comment:
Yes, but do you have a good idea how it could look like without breaking existing implementations? We cannot remove the existing methods from PNP (as that'd break existing implementations) so we only could add `getParameterNames(Executable e)`, but I think that'd only obfuscate the contract more than it helps.